app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -70,4 +73,21 @@ def not_found(options = nil)
 
     false
   end
+
+  def registration_url
+    uri = if params[:url].presence
+            URI.join(params[:url], '/register')
+          else
+            URI(register_url)
+          end
+
+    # Without protocol (but with port) uri.host returns nil
+    # URI(example.com:99).host => nil
+    unless uri.kind_of?(URI::HTTP) || uri.kind_of?(URI::HTTPS)
```

Comment:
please use `if` when the condition has either `||` or `&&`, it's much easier to read.

```
if !uri.kind_of?(URI::HTTP) && !uri.kind_of?(URI::HTTPS)
```

you could also do `if uri.scheme.start_with?('http')` or more explicit form `%w[http https].include?(uri.scheme)` which I personally find more readable and straightforward